### PR TITLE
Fix automation log branch highlighting

### PR DESCRIPTION
**Pull request by melohagan**
## Description
Fixes automation log flow highlighting issues where failed branch nodes and paths were not consistently highlighted, stopped paths could leak onto unexecuted branches, and multi-line log node labels shifted connector anchors out of alignment.

## Addresses
- https://github.com/Budibase/budibase/issues/18871

## Screenshots (Automation logs)

error path

automation log

## Launchcontrol
Fixes automation log branch and path highlighting so failed and stopped runs display the correct colours and aligned connectors.

---
## Summary by cubic
Fixes automation log branch, node, and edge highlighting so terminal failures, stopped-without-match runs, and executed paths use the right colors. Aligns handles in Logs and keeps selection outlines blue; default branch conditions now save reliably.

- Bug Fixes
  - Recognize “stopped without branch match” and show warn on branch wrappers, nodes, edges, and handles; Logs list shows Stopped. Errors still win when a terminal failure exists.
  - Treat terminal branch step failures as errors even if the branch didn’t progress; unexecuted branches stay unhighlighted.
  - Simplify primary branch edge routing (remove action-bar gap); align and color handles by state in Logs; selection always blue; add `BasicOperator.EQUAL` when an operator is missing; add tests for `BranchStepPanel` and update edge path tests.

<sup>Written for commit 2ad6343152bf479a795c410fafcfce9a99652f56. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/Budibase/budibase/pull/18873?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

File: packages/builder/src/components/automation/AutomationBuilder/FlowChart/FlowCanvas/nodes/NodeWrapper.svelte [hunk 1]
```diff
@@ -73,4 +105,17 @@
     width: fit-content;
     max-width: 360px;
   }
+  .step-wrapper.error :global(.custom-handle) {
```
Comment (adrinr) [inline packages/builder/src/components/automation/AutomationBuilder/FlowChart/FlowCanvas/nodes/NodeWrapper.svelte]: Why do we need this global? `custom-handle` is defined in here

File: packages/builder/src/components/automation/AutomationBuilder/FlowChart/FlowCanvas/edges/CustomEdge.svelte [hunk 2]
```diff
@@ -350,6 +363,19 @@
     )
   }
 
+  const didBranchStepFail = (branchStepId: string) => {
+    const results =
+      viewMode === ViewMode.LOGS
+        ? $automationStore.selectedLog
```
Comment (adrinr) [inline packages/builder/src/components/automation/AutomationBuilder/FlowChart/FlowCanvas/edges/CustomEdge.svelte]: Should this reactive statement be out of the function, in order to rerender when needed?

File: packages/builder/src/components/automation/AutomationBuilder/FlowChart/FlowCanvas/FlowEdgePaths.ts [hunk 3]
```diff
@@ -15,6 +15,7 @@ interface PrimaryBranchPathArgs {
   targetX: number
   targetY: number
   preBranchLabelX: number
+  reserveActionBarGap?: boolean
```
Comment (adrinr) [inline packages/builder/src/components/automation/AutomationBuilder/FlowChart/FlowCanvas/FlowEdgePaths.ts]: What does this mean?

File: packages/builder/src/components/automation/AutomationBuilder/FlowChart/FlowCanvas/nodes/NodeWrapper.svelte (hunk 1 above)
Comment (melohagan) [reply packages/builder/src/components/automation/AutomationBuilder/FlowChart/FlowCanvas/nodes/NodeWrapper.svelte]: Removed the global

File: packages/builder/src/components/automation/AutomationBuilder/FlowChart/FlowCanvas/edges/CustomEdge.svelte (hunk 2 above)
Comment (melohagan) [reply packages/builder/src/components/automation/AutomationBuilder/FlowChart/FlowCanvas/edges/CustomEdge.svelte]: Moved out of the function

File: packages/builder/src/components/automation/AutomationBuilder/FlowChart/FlowCanvas/FlowEdgePaths.ts (hunk 3 above)
Comment (melohagan) [reply packages/builder/src/components/automation/AutomationBuilder/FlowChart/FlowCanvas/FlowEdgePaths.ts]: Removed and code simplified

File: packages/builder/src/components/automation/AutomationBuilder/FlowChart/FlowCanvas/FlowEdgePaths.ts [hunk 4]
```diff
@@ -24,26 +24,17 @@ export const getPrimaryBranchPath = ({
...
   const radius = 12
 
   if (sourceY === targetY) {
-    return [
-      `M ${sourceX},${sourceY}`,
-      `L ${actionBarLeft},${sourceY}`,
-      `M ${actionBarRight},${sourceY}`,
-      `L ${targetX},${targetY}`,
-    ].join(" ")
+    return [`M ${sourceX},${sourceY}`, `L ${targetX},${targetY}`].join(" ")
```
[the remark this answers is not in the document]
Comment (melohagan) [reply packages/builder/src/components/automation/AutomationBuilder/FlowChart/FlowCanvas/FlowEdgePaths.ts:30]: The code was greatly simplified by letting the edge path go behind the action bar

**Review by adrinr — APPROVED** [review]
